service/backend/pchain/construction.go

```diff
@@ -62,15 +67,37 @@ func (b *Backend) ConstructionMetadata(
 		return nil, service.WrapError(service.ErrInvalidInput, err)
 	}
 
-	var suggestedFee *types.Amount
+	if opMetadata.Matches == nil {
+		return nil, service.WrapError(service.ErrInvalidInput, errors.New("matches not found in options"))
+	}
+
+	// Build a dummy base tx to calculate the base fee
+	dummyBaseTx, _, err := pmapper.BuildTx(
```

Comment:
Why do we have to use a dummy base tx instead of the actual type? Can we simply use opMetadata.Type for type and move this whole block after the switch case below, using the actual metadata?

This way you wouldn't have to buildTX twice for staking transactions for example.